common/lib/xmodule/xmodule/video_module/video_handlers.py

```diff
@@ -233,7 +266,8 @@ def transcript(self, request, dispatch):
 
             if language not in ['en'] + transcripts["transcripts"].keys():
                 log.info("Video: transcript facilities are not available for given language.")
-                return Response(status=404)
+                # Try to get transcript from edx-val.
+                return self.get_val_transcript(language)
 
             if language != self.transcript_language:
```

Comment:
@muhammad-ammar that's being set  in here as well. this will be current active transcript language.